sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/RestProxyImpl.java

```diff
@@ -365,26 +362,14 @@ private static Object handleRestResponseReturnType(Response<?> response, Swagger
                     responseBodyMode = requestOptions.getResponseBodyMode();
                 }
 
-                if (responseBodyMode == DESERIALIZE) {
-                    HttpResponseAccessHelper.setValue((HttpResponse<?>) response,
-                        handleResponseBody(response, methodParser, bodyType, response.getBody(), serializer));
-                } else {
-                    HttpResponseAccessHelper.setBodyDeserializer((HttpResponse<?>) response,
-                        (body) -> handleResponseBody(response, methodParser, bodyType, body, serializer));
-                }
-
-                Response<?> responseToReturn = createResponseIfNecessary(response, entityType, response.getBody());
-
-                if (responseToReturn == null) {
-                    return createResponseIfNecessary(response, entityType, null);
-                }
-
-                return responseToReturn;
+                // TODO (alzimmer): Need a way to support deferred deserialization.
+                return new Response<>(response.getRequest(), response.getStatusCode(), response.getHeaders(),
```

Comment:
If deferred serialization is used, then should we just return `Response<BinaryData>` here that is input to this method as that is a no-op here?